public-api/src/main/java/org/pmiops/workbench/publicapi/DataBrowserController.java

```diff
@@ -123,21 +156,121 @@
+                            .id(o.getId())
+                            .analysisId(o.getAnalysisId())
+                            .stratum1(o.getStratum1())
+                            .stratum2(o.getStratum2())
+                            .stratum3(o.getStratum3())
+                            .stratum4(o.getStratum4())
+                            .stratum5(o.getStratum5())
+                            .stratum5Name(o.getStratum5Name())
+                            .countValue(o.getCountValue());
                 }
             };
 
 
+    /* Final api functions put on top */
+    @Override
+    public ResponseEntity<DbDomainListResponse> getDomainFilters() {
+        List<DbDomain> domains=dbDomainDao.findByDbType("domain_filter");
+        DbDomainListResponse resp=new DbDomainListResponse();
+        resp.setItems(domains.stream().map(TO_CLIENT_DBDOMAIN).collect(Collectors.toList()));
+        return ResponseEntity.ok(resp);
+    }
+
+    @Override
+    public ResponseEntity<DbDomainListResponse> getSurveyList() {
+        List<DbDomain> domains=dbDomainDao.findByDbTypeAndAndConceptIdNotNull("survey");
+        DbDomainListResponse resp=new DbDomainListResponse();
+        resp.setItems(domains.stream().map(TO_CLIENT_DBDOMAIN).collect(Collectors.toList()));
+        return ResponseEntity.ok(resp);
+    }
+
+
+
+    @Override
+    public ResponseEntity<QuestionConceptListResponse> getSurveyResults(String surveyConceptId) {
+        /* Set up the age and gender names */
+        // Too slow and concept names wrong so we hardcode list
+        // List<Concept> genders = conceptDao.findByConceptClassId("Gender");
+        this.setAgeStratumNameMap();
+        this.setGenderStratumNameMap();
+        long longSurveyConceptId = Long.parseLong(surveyConceptId);
+        List<QuestionConcept> questions = questionConceptDao.findSurveyQuestions(longSurveyConceptId);
+        QuestionConceptListResponse resp = new QuestionConceptListResponse();
+        DbDomain survey = dbDomainDao.findByConceptId(longSurveyConceptId);
+        resp.setSurvey(TO_CLIENT_DBDOMAIN.apply(survey));
+
+        if (!questions.isEmpty()) {
+            for (int x = 0; x < questions.size() - 1; x++)  {
+                QuestionConcept q = questions.get(x);
+                String qid = Long.toString(q.getConceptId());
+
+                /*** Important -- when calling these dao functions in a loop, baceause the analyses share same id ,
+                 * and we are filtering by a property in the related reluslts table, the first results get cached and
+                 * put on all questions. So we have to detach the entity from the session
+                 */
+                List<AchillesAnalysis> analyses = achillesAnalysisDao.findQuestionAnalysisResults(surveyConceptId, qid);
+                //THIS IS THE IMPORTANT PART
+                //You have to detach the entity from the session otherwise all the results are same as the first
```

Comment:
I don't understand this. Can you elaborate on what happens and why?

Reply:
So calling this in the the for each question loop, causes the achillesAnalysisDao.findQuestionAnalysisResults(...) to cache the analyses  of the first question in the loop. So next loop iteration the question is different, the argument to the function is different, but the analyses list is the same. It has to do with the analysis objects having same id for each question. Level 1 caching in Hibrenate ... from the stackoverflow post: 

  You have to detach the entities from the session when the results returned may have collisions on the @Id column. This has to do with the way level 1 caching works in Hibernate. You can check out this link for a little more info http://docs.jboss.org/hibernate/entitymanager/3.6/reference/en/html/objectstate.html